### ISSUE TITLE
Support projects which has phases but doesn't have timelines

### ISSUE THREAD

**Issue by maxceem** [appirio-tech/connect-app#2308]
There are some projects which have phases but doesn't have timelines. See for example https://connect.topcoder-dev.com/projects/6375/plan

To support them properly:
1. Remove timeline tab:
    
    ![image](https://user-images.githubusercontent.com/146016/43778125-71c11830-9a87-11e8-8cb0-5f4acd0d79c5.png)

2. Fix error in browser console:
    ```
     productsTimelines.js:112 Uncaught (in promise) TypeError: Cannot read property 'milestones' of undefined
    at productsTimelines (productsTimelines.js:112)
    at combination (combineReducers.js:124)
    at computeNextEntry (<anonymous>:1:33861)
    at recomputeStates (<anonymous>:1:34161)
    at <anonymous>:1:37746
    at Object.dispatch (createStore.js:169)
    at dispatch (<anonymous>:1:38223)
    at eval (index.js:104)
    at handleAction (index.js:150)
    at eval (index.js:63)
    ```

**Comment by maxceem** [comment]
Done.

3. Previously timelines were loaded by `ProductTimelinesContainer` which was mounted inside the timeline tab. But now we need to know if we have to show the tab in advance because we have to decide to show it or not. So I've moved loading timelines to the dashboard action. So now timelines are being loaded for the whole page together with the rest of the stuff.
As we actually need timelines data not only inside timeline container but also to decide show/hide timeline tab itself and to calculate phases durations and dates.

**Comment by vikasrohit** [comment]
Thanks @maxceem. Yes, this should be intended behaviour and in fact I was under impression we got it handled already. Anyway, it is good to get it fixed before release.
On your last point, I understand that we need to load the timelines data on dashboard to be able to update the UI portions based on that. I just want to make sure that, as we are not showing the timeline data by default on page load, the call to fetch timelines, could work silently behind the scene and most probably by the time user interacts with UI, it has the data and already update the UI without showing any loading icon.

**Comment by maxceem** [comment]
@vikasrohit after this update timeline data IS included in project loading indicator. Means loading timelines prolongs the whole project loading time after this fix. 

I implemented it this way for now as we also need timeline data to calculate phases durations and start/end date.
If we make loading timelines silent then by default phases will show dates and duration based on their internal data. After timelines are silently loaded start/end time and duration will be updated at phases cards. Which feels like a glitch as they are updated without any visible reasons for a user.

Let me know what you think.

**Comment by vikasrohit** [comment]
Yup, makes sense. Lets stick to the way you implemented right now. But we need to find out way to improve the dashboard load performance.
